app/helpers/fenrir_view/component_helper.rb

```diff
@@ -1,23 +1,63 @@
+# frozen_string_literal: true
+
 module FenrirView
   module ComponentHelper
+
+    # TODO: Remove?
     def render_ui(variant, slug, properties = {}, &block)
       return nil if properties == false
 
       component = FenrirView::Presenter.component_for(variant, slug, properties, validate: true)
+
+      if block_given?
+        if defined?(component.layout)
```

Comment:
Might want to use `component.respond_to?(:layout)` which is basically equivalent but a bit more Ruby-duck-typish.